### PR TITLE
[3.3.2] Fields length validation

### PR DESCRIPTION
**Pull request by ViacheslavKlimov**

### CODE REVIEW

File: common/data/src/main/java/org/thingsboard/server/common/data/widget/WidgetsBundle.java [hunk 1]
```diff
@@ -31,6 +30,7 @@
     @NoXss
     private String alias;
```
Comment (ViacheslavKlimov) [inline common/data/src/main/java/org/thingsboard/server/common/data/widget/WidgetsBundle.java:31]: Alias length should also be less than 255

File: common/data/src/main/java/org/thingsboard/server/common/data/widget/WidgetsBundle.java [hunk 2]
```diff
@@ -31,6 +30,7 @@
...
     @NoXss
+    @Length(fieldName = "title")
     private String title;
     private String image;
     @NoXss
```
Comment (ViacheslavKlimov) [inline common/data/src/main/java/org/thingsboard/server/common/data/widget/WidgetsBundle.java:36]: Same as for alias

File: common/data/src/main/java/org/thingsboard/server/common/data/User.java [hunk 3]
```diff
@@ -36,8 +36,10 @@
     private String email;
     private Authority authority;
     @NoXss
+    @Length(fieldName = "firs name")
```
Comment (ViacheslavKlimov) [inline common/data/src/main/java/org/thingsboard/server/common/data/User.java:39]: Typo

File: dao/src/main/java/org/thingsboard/server/dao/service/StringLengthValidator.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+@Slf4j
+public class StringLengthValidator implements ConstraintValidator<Length, String> {
+    private int max;
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        if (StringUtils.isEmpty(value)) {
+            return true;
+        }
+        return value.trim().length() <= max;
```
Comment (ViacheslavKlimov) [inline dao/src/main/java/org/thingsboard/server/dao/service/StringLengthValidator.java:34]: We need to check the length of untrimmed value here as we will then save it as is and sql error will occur

File: application/src/main/java/org/thingsboard/server/exception/ThingsboardErrorResponseHandler.java [hunk 5]
```diff
@@ -71,6 +74,17 @@ public void handle(Exception exception, HttpServletResponse response) {
...
+                String cause = "";
+                if (exception.getCause() != null) {
+                    cause = exception.getCause().getClass().getCanonicalName();
+                }
+                if (exception instanceof IllegalArgumentException || exception instanceof IncorrectParameterException
+                        || exception instanceof DataValidationException || cause.contains("IncorrectParameterException")) {
+                    exception = new ThingsboardException(exception.getMessage(), ThingsboardErrorCode.BAD_REQUEST_PARAMS);
+                } else if (exception instanceof MessagingException) {
+                    exception = new ThingsboardException("Unable to send mail: " + exception.getMessage(), ThingsboardErrorCode.GENERAL);
+                }
```
Comment (ViacheslavKlimov) [inline application/src/main/java/org/thingsboard/server/exception/ThingsboardErrorResponseHandler.java:86]: Exceptions from controllers will not be logged according to BaseController.handleThingsboardException logic
```
if (logException && logControllerErrorStackTrace) {
    log.error("Error [{}]", exception.getMessage(), exception);
}
```

**Review by vvlladd28 — APPROVED** [review]

**Pull request merged**